Mage.Sets/src/mage/cards/p/ProgenitorsIcon.java

```diff
@@ -0,0 +1,148 @@
+        // Based on Circle of Solace
+        this.addAbility(new AsEntersBattlefieldAbility(new ChooseCreatureTypeEffect(Outcome.Neutral)));
+
+        // {T}: Add one mana of any color.
+        this.addAbility(new AnyColorManaAbility());
+
+        // {T}: The next spell of the chosen type you cast this turn can be cast as though it had flash.
+        // Based on Ride the Avalanche
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new ProgenitorsIconAsThoughEffect(), new TapSourceCost());
+        ability.addWatcher(new ProgenitorsIconWatcher());
+        this.addAbility(ability);
+    }
+
+    private ProgenitorsIcon(final ProgenitorsIcon card) {
+        super(card);
+    }
+
+    @Override
+    public ProgenitorsIcon copy() {
+        return new ProgenitorsIcon(this);
+    }
+}
+
+class ProgenitorsIconAsThoughEffect extends AsThoughEffectImpl {
+
+    public ProgenitorsIconAsThoughEffect() {
+        super(AsThoughEffectType.CAST_AS_INSTANT, Duration.EndOfTurn, Outcome.Benefit);
+        staticText = "The next spell of the chosen type you cast this turn can be cast as though it had flash";
+    }
+
+    private ProgenitorsIconAsThoughEffect(final ProgenitorsIconAsThoughEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public void init(Ability source, Game game) {
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+        ProgenitorsIconWatcher.addPlayer(source.getControllerId(), subType, game);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public ProgenitorsIconAsThoughEffect copy() {
+        return new ProgenitorsIconAsThoughEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+
+        if (!ProgenitorsIconWatcher.checkPlayerCast(affectedControllerId, subType, game)) {
+            discard();
+            return false;
+        }
+        Card card = game.getCard(sourceId);
+        if (card == null){
+            return false;
+        }
+
+        return card.getSubtype().contains(subType);
+    }
+}
+
+class ProgenitorsIconWatcher extends Watcher {
+
+    // Handles multiple instances of this same card choosing the same or different subtypes for the same player
+    private final Map<UUID, ArrayList<SubType>> playerSubTypesChosen = new HashMap<>();
+
+    public ProgenitorsIconWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.SPELL_CAST){
+            UUID playerID = event.getPlayerId();
```

Comment:
personally I'd just inline this method call the one place it's used

Reply:
I'm not sure what you mean by this--the watch() method signature has to match the one its overriding, unless you mean something else.

Reply:
ah sorry I meant `event.getPlayerId()` which you can just call where you need it, rather than storing to `UUID playerId`